sigstore-java/src/main/java/dev/sigstore/bundle/Bundle.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2024 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.bundle;
+
+import com.google.common.base.Preconditions;
+import dev.sigstore.KeylessSignature;
+import dev.sigstore.rekor.client.RekorEntry;
+import java.io.Reader;
+import java.security.cert.CertPath;
+import java.util.List;
+import java.util.Optional;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Lazy;
+
+@Immutable
+public interface Bundle {
+
+  enum HashAlgorithm {
+    SHA2_256
+  }
+
+  String BUNDLE_V_0_1 = "application/vnd.dev.sigstore.bundle+json;version=0.1";
+  String BUNDLE_V_0_2 = "application/vnd.dev.sigstore.bundle+json;version=0.2";
+  String BUNDLE_V_0_3 = "application/vnd.dev.sigstore.bundle+json;version=0.3";
+  // media_type format switch: https://github.com/sigstore/protobuf-specs/pull/279
+  String BUNDLE_V_0_3_1 = "application/vnd.dev.sigstore.bundle.v0.3+json";
+  List<String> SUPPORTED_MEDIA_TYPES =
+      List.of(BUNDLE_V_0_1, BUNDLE_V_0_2, BUNDLE_V_0_3, BUNDLE_V_0_3_1);
+
+  String getMediaType();
+
+  Optional<MessageSignature> getMessageSignature();
+
+  Optional<DSSESignature> getDSSESignature();
+
+  @Value.Check
+  default void checkOnlyOneSignature() {
```

Comment:
are these javadoc worthy? I guess they're for Immutables, but they'll show up in the API anyway.  Maybe just a comment describing why they're there?

Reply:
I think this section needs to be protected or something. hidden in an abstract class rather than an interface